library/threading.c

```diff
@@ -30,15 +30,15 @@
 #include "mbedtls/threading.h"
 
 #if defined(MBEDTLS_THREADING_PTHREAD)
-static void threading_mutex_init_pthread( mbedtls_threading_mutex_t *mutex )
+void mbedtls_mutex_init( mbedtls_threading_mutex_t *mutex )
 {
     if( mutex == NULL )
```

Comment:
I'm not certain if these tests are necessary. Adding them not only introduces branches but also enforces virtually repeating the same test in the client code as for each of the calls to this API the client must check for the error code. It may be better to put the responsibility of providing valid mutexes on the client an only check for the validity once instead of every time these functions are called.

Reply:
Do you mean the parameter validation checks? eg. 
```
if( mutex == NULL )
{ 
...
}
```

We've had strong feedback from some users that their absence on many functions is poor practice, and they want to see a lot more. Conversely, some users see them as redundant and unnecessary and taking up precious code space. (And there's another camp that thinks they should be `asserts`).

I think the code can stay as it is for the moment (eg. no change), but at some point we'll need to review the library and introduce more checks but make them optional at compile time (similar to asserts).